Atulk/11621 sweep pr

yugi957 opened the pull request:
### Ticket
#11621 

### Problem description
 - Sweeps for many tm ops, squeeze, unsqueeze, embedding, view, repeat, split

### What's changed
 - Created sweeps for squeeze, unsqueeze, embedding, view, repeat, split

### Checklist
- [x] Post commit CI: https://github.com/tenstorrent/tt-metal/actions/runs/11488803697

shwetankTT committed:
#10110: conv non tile sharding

yugi957 committed:
#13181: added sweep for squeeze_pytorch2

yugi957 committed:
#13181: added sweep for squeeze_pytorch2

yugi957 committed:
#13181: sweep only for model traces

yugi957 committed:
#13181: sweep only for model traces

yugi957 committed:
#11621: merged yaml changes

yugi957 committed:
#11621: merged yaml changes

yugi957 committed:
#11621: merged embedding sweep changes

yugi957 committed:
#11621: merged embedding sweep changes

yugi957 committed:
#11621: merging view changes

yugi957 committed:
#11621: merging view changes

yugi957 committed:
#11621: merging sweep changes

yugi957 committed:
#11621: merging sweep changes

umadevimcw committed:
#8593: Fix pow pcc drop issue

mrshaw01 committed:
#14087: Support bfloat8_b for get_lib_dtype test_utils.py

mrshaw01 committed:
#14087: Add bfloat8_b unit test for moreh_arange

mrshaw01 committed:
#14087: Use ttnn.to_torch, from_torch in unit test moreh bmm (forward, backward)

mrshaw01 committed:
#14087: Use float value, modulize unit test moreh_bmm (forward, backward)

mrshaw01 committed:
#14087: Add bfloat8_b unit test for moreh_bmm (forward, backward)

mrshaw01 committed:
#14087: Modulize unit test for moreh_norm (forward, backward)

mrshaw01 committed:
#14087: Add bfloat8_b unit test for moreh_norm (forward, backward)

mrshaw01 committed:
#14087: Use logger.info instead of logger.debug

mrshaw01 committed:
#14087: Disable float32 accumulation for Grayskull

mrshaw01 committed:
#14087: Optimize get_tensors for unit test moreh bmm (forward, backward)

mrshaw01 committed:
#14087: Enable Wormhole B0 compute kernel configuration in unit testing properly

yugi957 committed:
#11621: add squeeze workflow

yugi957 committed:
#11621: add squeeze workflow

blozano-tt committed:
[skip-ci] #0: Use PR Comment to trigger test workflow (#14184)

jdesousa-TT commented on tests/sweep_framework/sweeps/data_movement/embedding/embedding_pytorch2.py:
```diff
@@ -0,0 +1,211 @@
+    return brackets_content
+
+
+def parse_md_file_simple_no_regex(file_path):
```
Is this used in embedding at all?

yugi957 replied:
I think this is actually not, maybe we can take it out of this trace

jdesousa-TT commented on tests/sweep_framework/sweeps/data_movement/unsqueeze/unsqueeze_pytorch2.py:
```diff
@@ -0,0 +1,318 @@
+# dim = i or -(length-i) 50% chance here
+# All can go in a for loop for how many samples you want
+# This will do every combination of the following dtype and layout, so we will address RM and bfloat8 in invalidate vector
+def generate_unsqueeze_config(num_samples=10):
```
Is this used?

jdesousa-TT commented on tests/sweep_framework/sweeps/data_movement/squeeze/squeeze_pytorch2.py:
```diff
@@ -0,0 +1,119 @@
+# dim = i or -(length-i) 50% chance here
+# All can go in a for loop for how many samples you want
+# This will do every combination of the following dtype and layout, so we will address RM and bfloat8 in invalidate vector
+def generate_squeeze_config(num_samples=10):
```
I see this is commented out in parameters... are all of these the functions that we're used to generate the params? Leaving the code for future reference?

yugi957 replied:
I'm leaving all these generation functions in for when we do adversarial testing!

jdesousa-TT commented on tests/sweep_framework/sweeps/data_movement/view/view_pytorch2.py:
```diff
@@ -0,0 +1,121 @@
+
+parameters = {
+    "nightly": {
+        "view_specs": parse_md_file_simple_no_regex("sweeps/data_movement/view/view_trace.md"),
```
We'll have to have this file checked in for this to run right?

sjameelTT replied:
Yeah can we copy and paste it in here? If you want to update it in the future when PT2.0 people change things around then you can do some follow-ups.

jdesousa-TT replied:
Should be able to just drop it in that folder yes

yugi957 replied:
Yup, for now just copy paste md file in. I was talking with Kevin I believe about how we should move forward. JSON is most likely best if we can get it.

jdesousa-TT replied:
Can you drop in the file then we can merge this one. I just don't want to merge without the file because it will break CI.

yugi957 replied:
Have to make a new pr because of some merge I messed up with, but that one will contain the md

blozano-tt committed:
[skip ci] #0: Call all-post-commit on comment (#14187)

blozano-tt committed:
Update test-comment.yaml

blozano-tt committed:
#11830: Remove unnecessary includes (#14131)

sjameelTT reviewed (commented):
Could you run each of the sweeps using the sweep run action?

Aswinmcw committed:
#14095: Move ccl unit tests to separate directory (#14096)

BuiChiTrung committed:
#13320: Impl uniform operation in ttnn (#13735)

* #13320: add draft code to test dropout op

#13320: rollback add_2_integers_in_compute changes

#13320: add draft code for uniform in programming_examples

#13320: check random numbers frequency using map

#13320: add draft code

#13320: try remove reader kernel and cb1

#13320: add skeleton code for uniform

#13220: support from, to param

#13320: remove reader kernel

#13320: add cb_intermed0

#13320: add unit-test

#13320: refactor writer kernel by exposing an sfpu API

#13320: remove native dir

#13320: refactor

#13320: update unit test

#13320: change compute_kernel_config param to const optional and add license

#13320: add cb id as compile args

#13320: raise exception when from = to

#13320: skip test for grayskull

#13320: revise uniform operation

#13320: update program factory

#13320: update unit-test for bfloat16 and refactor cb var name in kernels

#13320: add feature to benchmark ttnn with torch uniform

#13320: refactor

#13320: update compute_output_shapes func

* #13320: update unit-test

* #13320: update sfpu api

* #13320: remove un-used header in kernel and counter in sfpu API

* #13320: remove TTI_SFPSETSGN instr

* #13320: allow set fp32_dest_acc to false

* #13320: update callback test

* #13320: skip test for grayskull

* #13320: update sfpu to generate random float

* #13320: update sfpu to generate random float between range [from, to)

* #13320: refactor writer kernel

* #13320: update compute kernel

* #13320: update sfpu to use TTI_ only

* #13320: change cb format to float32

* #13320: update var name

* #13320: draft commit to support generate random bfloat16 in sfpu

* #13320: rollback to inital approach to handle bfloat16

* #13320: rebase with main

* #13320: rm unused header

mrshaw01 committed:
#14208: Refactor and add bfloat8_b unit test moreh_mean (forward, backward)

mcw-anasuya committed:
#8343: Update documentation for ttnn.glu, reglu, geglu, swiglu (#13932)

* #8343: Update documentation and move sweep test

* #8343: Remove timeout parameter

mcw-anasuya committed:
#13674: Update documentation for acosh and other bw ops (#14161)

* #13674: Update documentation for backward ops

* #13674: Fix bw acosh and acos description

Commit:
#8142: TTNN log-alike ops incomplete documentation (#13969)

* #8142: Add documentation for log alike ops

* #8142: Update sweep tests

rdjogoTT committed:
#13653: add .md for reconfig data format (#13988)

* #13653: add .md for reconfig data format

* #13653: add supported reconfigs to .md

yugaoTT committed:
#0: fix watcher error on cb ptr overflow (#14162)

tt-aho committed:
#0: [skip-ci] Add introduction section to model perf optimizations tech report (#14220)

Rename images and folder for model optimizations tech report

sjameelTT commented:
> Could you run each of the sweeps using the sweep run action?

Actually this may not be necessary

KalaivaniMCW committed:
#0: update documentation and asserts (#14101)

#13928: update binary bw doc

ntarafdar committed:
Fix Bug in Reshape where Output Tensor is Different Layout than Input Tensor (#14176)

* #13891: fix bug in reshape that returns different tensor layout than input

* #0: rename row_major_reshape to reflect that layout is converted to orignal layout

amalbasaTT committed:
reduction ops sweeps: argmax and topk (#14221)

* #11512: Add topk sweep

* #11512: Add argmax sweep, update ttnn-run-sweeps.yaml file

* #11512: Add golden_function usage in reduction sweeps

* #11512: Modify invalidate_vector in argmax.py

* #11512: add bfloat8_b to argmax and fix tt_smi_util import in sweeps_runner

ntarafdar committed:
Fix Hanging Reshape (#14120)

#13799: fix hanging reshape

ntarafdar committed:
Update CODEOWNERS (#14235)

[skip ci] #0: Update CODEOWNERS to reflect current data_movement team

nathan-TT committed:
#0: Kill l1_test_data_only (#14219)

sminakov-tt committed:
#0: Unify `tt_metal::Buffer` and `ttnn::Buffer`, add support for 0-volume buffers (#14023)

* #0: Buffer refactoring wip

* #0: Buffer threading change

* #0: Buffer fixup

* #0: Buffer fixup

* #0: Add experimental asserts

* #0: Refactor

* #0: Test fixup

* #0: Review fixes

* #0: Review fixes

* #0: Review fixes

* #0: Handle 0 page size

* #0: Add a test for creating/reading/writing 0-sized tensor

* #0: Add support for more 0-sized shapes

* #0: Review fixes

* #0: Refactor allocation/deallocation threading to allow getting buffer address from another thread

* #0: Treat empty shape as scalar

* #0: Fix trailing comma

* #0: Add missing include

* #0: Allow 0 page size

* #0: Rewrite threading, no mutexes

* #0: Assert fix

* #0: Fixup

* #0: Remove unneeded include

* #0: Cleanup, fix typo

* #0: Cleanup

* #0: Trying to improve perf

* #0: Trying to improve perf

* #0: Build fix

* #0: Simpler threading, get rid of canceling allocation feature

* #0: Fixup

* #0: Avoid using std::atomic::wait, it seems to have perf implications

---------

Co-authored-by: Artem Yerofieiev <169092593+ayerofieiev-tt@users.noreply.github.com>

ntarafdar committed:
#13889: 5d reshape fixed

ntarafdar committed:
#13889: check for output shape rank

yugi957 committed:
#11621: view_trace md

yugi957 committed:
#11621: merge commit

SeanNijjar commented:
Can you please reopen this PR or fix the branch because it pulled in lots of other unrelated changes that are polluting the PR.

pavlejosipovic commented on ttnn/cpp/ttnn/operations/conv/conv2d/device/kernels/conv_bmm_tilize_col_major_out_blocks.cpp:
```diff
@@ -46,22 +46,24 @@ inline void tilize_in(
     uint32_t interm_cb_id,
     uint32_t out_cb_id) {
-
+    constexpr bool is_non_tile_height_= is_non_tile_height;
```
This seems like bad rebase/merge?
